Snowflake.Data.Tests/IntegrationTests/SFDbCommandIT.cs

```diff
@@ -175,6 +175,289 @@ public void TestExecuteAsyncWithMaxRetryReached()
                 Assert.GreaterOrEqual(stopwatch.ElapsedMilliseconds, 30 * 1000);
             }
         }
+
+        [Test]
+        public async Task TestAsyncExecQueryAsync()
+        {
+            string queryId;
+            var expectedWaitTime = 5;
+
+            using (SnowflakeDbConnection conn = new SnowflakeDbConnection())
+            {
+                conn.ConnectionString = ConnectionString;
+                await conn.OpenAsync(CancellationToken.None).ConfigureAwait(false);
+
+                using (SnowflakeDbCommand cmd = (SnowflakeDbCommand)conn.CreateCommand())
+                {
+                    // Arrange
+                    cmd.CommandText = $"CALL SYSTEM$WAIT({expectedWaitTime}, \'SECONDS\');";
+
+                    // Act
+                    queryId = await cmd.ExecuteAsyncInAsyncMode(CancellationToken.None).ConfigureAwait(false);
+                    var queryStatus = await cmd.GetQueryStatusAsync(queryId, CancellationToken.None).ConfigureAwait(false);
+
+                    // Assert
+                    Assert.IsTrue(QueryStatuses.IsStillRunning(queryStatus));
+
+                    // Act
+                    DbDataReader reader = await cmd.GetResultsFromQueryIdAsync(queryId, CancellationToken.None).ConfigureAwait(false);
+                    queryStatus = await cmd.GetQueryStatusAsync(queryId, CancellationToken.None).ConfigureAwait(false);
+
+                    // Assert
+                    Assert.IsTrue(reader.Read());
+                    Assert.AreEqual($"waited {expectedWaitTime} seconds", reader.GetString(0));
+                    Assert.AreEqual(QueryStatus.Success, queryStatus);
+                }
+
+                await conn.CloseAsync(CancellationToken.None).ConfigureAwait(false);
+            }
+        }
+
+        [Test]
+        public async Task TestMixedSyncAndAsyncQueryAsync()
+        {
+            string queryId;
+            var expectedWaitTime = 5;
+
+            // Start the async exec query
+            using (SnowflakeDbConnection conn = new SnowflakeDbConnection())
```

Comment:
you are starting 3 connections one by one and closing them. But when you close the connection the session is returned to the pool so the next connection probably uses the same session. Do we want to demonstrate that they are separate sessions? If so then we should open 3 connections at the begining.

Reply:
It's more about showing a normal query can be ran while an async exec query is already running. But using different sessions can be included in the test as well just to show that it works

Modified test to use different sessions